### ISSUE TITLE
[question] conan.tools.cmake.CMake + CMakeDeps = missing CMakePresets.json

### ISSUE THREAD

**Issue by n00b42** [conan-io/conan#12896]
### What is your question?

- Previously, I used `conans.CMake` in combination with `CMakeDeps`.
- It seems this combination was removed in Conan 1.57, breaking my build.
- As suggested I switched to `conan.tools.cmake.CMake`
- Running `conan create [..]` I get `FileNotFoundError [...] CMakePresets.json` with the following recipe:
~~~
from conan.tools.cmake import CMake, CMakeDeps
class Recipe(ConanFile):
	requires = [ ... ]
	settings = { "os", "compiler", "build_type", "arch" }
	name = "hello"
	version = "0.1"
	def generate(self):
		cmake = CMakeDeps(self)
		cmake.generate()
	def build(self):
		cmake = CMake(self)
~~~
- What am I missing?

System: Debian Testing, conan 1.57, cmake version 3.25.1

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @n00b42 

Thanks for your question.
In Conan 1.57 we changed a bit the ``cmake_layout()`` because there was a serious bug that affects single-config cmake generation (like the one in Linux "Unix Makefiles"). Now, CMakeToolchain generated files like ``CMakePresets.json`` will be located inside ``build/Release/generators``, instead of ``build/generators``. 
But that shouldn't be affecting the standard integrations, so there might be something else missing. Could you please post a more complete recipe to reproduce? Maybe you are calling ``cmake --presets`` yourself? I am missing the ``CMakeToolchain`` in your recipe too, which is necessary to work with the new ``CMake`` integration.

**Comment by AbrilRBS** [comment]
Hi! Thanks for taking the time to report your issue

Could you try adding `generators = "CMakeToolchain"` to your recipe? This generator is needed for cases like this and should fix the error you're seeing.

What we can do is to improve the UX around this, even if it's just having a prettier/more descriptive error message :)

**Comment by n00b42** [comment]
@memsharded 
I put together an example that worked in 1.56.0, and another example how I tried to fix it for 1.57.0.
- https://github.com/n00b42/conan-issue-12896-example
- The repos should contain all necessary files, and the README shows the conan calls I used.

**Comment by n00b42** [comment]
@RubenRBS added a third example adding `CMakeToolchain` as additional generator as suggested and it worked. Thank you very much. Added it as "new" in my example repo for now.

**Comment by memsharded** [comment]
Thanks for the repro case!
I have validated that it fails (ugly) and indeed
```diff
diff --git a/NEW/conanfile.py b/NEW/conanfile.py
index 3661120..746e250 100644
--- a/NEW/conanfile.py
+++ b/NEW/conanfile.py
@@ -7,7 +7,7 @@ class Recipe(ConanFile):
        version = "0.1"

        exports_sources = "*"
-       generators = "CMakeDeps"
+       generators = "CMakeDeps", "CMakeToolchain"

        def build(self):
                cmake = CMake(self)
```
fixes the issue.

 I agree we need to improve a bit the UX, giving a meaningful error instead of the FileNotFound, so lets assign this to next release. Thanks for the feedback!

**Comment by franramirez688** [comment]
Closed by https://github.com/conan-io/conan/pull/13019

It'll be released in the next Conan 1.59.
